scripts/benchmark_db/bench.rb

```diff
@@ -76,8 +74,42 @@ def sample_proc(pid, metrics)
   metrics[:vsz].push(output[1].to_i)
 end
 
-def proc_monitor(pid)
-  metrics = { rss: [], vsz: [] }
+def get_last_epoch(benchmark)
+  epoch = nil
+  loop do
+    epoch = benchmark.epoch_command
+    sleep 0.5
+    # epoch can be nil (e.g. if client is in "fetchting messages" stage)
+    break if epoch
+  end
+  epoch
+end
+
+def measure_online_validation(benchmark, pid, metrics)
```

Comment:
Some method comments would be nice, especially when the method is mutating its inputs.